### PR TITLE
#[164046234] write unit tests for implemented functionality

### PR DESCRIPTION
**Pull request by MandelaK**
### What does this PR do?
- write unit tests for models
- write unit tests for views
- write unit tests for serializers
- install coverage to get coverage and create `.coveragerc` file to configure ignored files

### Description of the task to be completed
The codebase currently includes implemented functionality for the project. We wrote tests that assert that all currently implemented functionality works without changing any of the code. We also wrote integration tests for the views. 

### How can this be manually tested?
- Clone this repository using `git clone https://github.com/andela/ah-legion-backend.git`
- Checkout to this branch using `git checkout ch-initial-unit-tests-164046234`
- Make sure to install coverage, it is included in the requirements.txt file.
- Make sure you have your environment variables configured in a .env file 
- When in the root directory, make migrations using `python manage.py makemigrations --settings=authors.settings.test` and then run `python manage.py migrate --settings=authors.settings.test`
- Run tests using `coverage run manage.py test --settings=authors.settings.test`
- To get coverage, you can run the command `coverage report`

### Any background context you want to provide?
We only wrote tests for currently implemented features and did not alter any functionality.

### What are the relevant PT stories?
[#164046234](https://www.pivotaltracker.com/story/show/164046234)

### CODE REVIEW

**Review by cob04 — DISMISSED** [review]
tests run, coverage is at 97%

**Review by Etomovich — DISMISSED** [review]
The test seem to be running well good to see.

**Review by ElMonstro — DISMISSED** [review]
Good test coverage

**Review by BrianSerem — DISMISSED** [review]
Looks good, good job with the tests

**Review by RachelleMaina — COMMENTED** [review]
python-decouple is not in the requirements.txt. I also get this error `ValueError: Dependency on app with no migrations: authentication`

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+omit =
+   *migrations*
+   *tests*
+   manage.py
+   *settings*
+
+branch = True
+
+[report]
+show_missing = True
```
Comment (abulojoshua1) [inline .coveragerc]: @MandelaK, please add an End of File here.

File: authors/settings/base.py [hunk 2]
```diff
@@ -123,4 +123,4 @@
 REST_FRAMEWORK = {
     'EXCEPTION_HANDLER': 'authors.apps.core.exceptions.core_exception_handler',
     'NON_FIELD_ERRORS_KEY': 'error',
-}
+}
```
Comment (abulojoshua1) [inline authors/settings/base.py]: Please add End of File here.

File: authors/apps/authentication/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        update_info = {"user": {
+            "email": "user1@mail.com",
+            "bio": "I love to be tested",
+            "values": "EPIC"
+        }}
+        client = APIClient()
+        client.post(reverse('authentication:register'), user1, format='json')
+        client.login(email="user1@mail.com", password="password")
+        response = client.put(reverse('authentication:get users'), update_info, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (abulojoshua1) [inline authors/apps/authentication/tests/test_views.py]: Please add End of File here.

**Review by BrianSerem — APPROVED** [review]
Looks fine

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Review by Etomovich — APPROVED** [review]
Nice job, Keith your build is working nice job!!

File: authors/apps/authentication/tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        """Test if proper string representation is returned for users"""
+        self.assertEqual(str(self.user1), "user@mail.com")
+
+    def test_get_full_name(self):
+        """Test if proper full name is returned for users"""
+        self.assertEqual(self.user1.get_full_name, "user")
+
+    def test_get_short_name(self):
+        """Test if proper short name is returned for users"""
+        self.assertEqual(self.user1.get_short_name(), "user")
```
Comment (abulojoshua1) [inline authors/apps/authentication/tests/test_models.py]: @MandelaK, please add EOF here.

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/tests/test_models.py (hunk 4 above)
Comment (MandelaK) [reply authors/apps/authentication/tests/test_models.py]: Ok, done

**Review by cob04 — APPROVED** [review]
Travis in the green

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**